File: 3914-check-if-any-element-has-prime-frequency/check-if-any-element-has-prime-frequency.java
import java.util.*;

class Solution {
    public boolean checkPrimeFrequency(int[] nums) {
        List<Integer> frequencies = new ArrayList<>();
        
        Map<Integer, Integer> freqMap = new HashMap<>();
        for (int num : nums) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
        }
        
        frequencies.addAll(freqMap.values());
        
        Set<Integer> uniqueFreqs = new HashSet<>(frequencies);
        
        for (int freq : uniqueFreqs) {
            if (isPrime(freq)) {
                return true;
            }
        }
        
        return false;
    }
    
    private boolean isPrime(int n) {
        if (n <= 1) {
            return false;
        }
        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: 3914-check-if-any-element-has-prime-frequency/check-if-any-element-has-prime-frequency.py
class Solution:
    def checkPrimeFrequency(self, nums: List[int]) -> bool:
        l = []
        def is_prime(n):
            if n <= 1:
                return False
            for i in range(2, int(n ** 0.5) + 1):
                if n % i == 0:
                    return False
            return True
        for i in nums:
            l.append(nums.count(i))
        for i in l:
            if is_prime(i):
                return True
        return False